Add experimental support to build using Bikeshed

foolip opened the pull request:
The new `--bikeshed` flag enables Bikeshed mode. The wattsi2bikeshed.js script converts the source from Wattsi to Bikeshed syntax.

The hardest part of this is ensuring that cross-references still resolve to the same thing. Wattsi's `<span>` and `<i>` for autolinks are converted to `<a>`, but changes to HTML itself will likely be needed for some cases where the `data-x=""` (with hyphens) is unambiguous but where Bikeshed's linking text (with spaces) is ambiguous and would require a `for=""` attribute.

What doesn't work:

  - Some cross-references as per above.
  - Some manual changes to HTML are still needed:
    https://github.com/whatwg/html/tree/bikeshed-experiment
  - Existing IDs aren't always preserved.
  - Simplification of HTML into Bikeshed shorthands like `{{Node}}`.
  - `subdfn`, which has no equivalent in Bikeshed.
  - Variants like the dev edition aren't properly supported.

foolip commented on build.sh:
```diff
@@ -663,27 +671,33 @@ function processSource {
-    fi
+  if [[ $USE_BIKESHED == "true" ]]; then
+    echo "BIKESHED!!!"
+    bikeshed spec --byos "$HTML_TEMP/source-whatwg-complete" "$HTML_TEMP/bikeshed-output" --md-Text-Macro="SHA $HTML_SHA"
```
Not sure if `--byos` (bring your own spec) helps here, didn't try with and without.

annevk commented:
Happy to rubber stamp this, but let's give @sideshowbarker and @zcorpan a chance to chime in as well.

foolip committed:
Add experimental support to build using Bikeshed

The new --bikeshed flag enables Bikeshed mode. The wattsi2bikeshed.js
script converts the source from Wattsi to Bikeshed syntax.

The hardest part of this is ensuring that cross-references still resolve
to the same thing. Wattsi's <span> and <i> for autolinks are converted
to <a>, but changes to HTML itself will likely be needed for some cases
where the data-x="" (with hyphens) is unambiguous but where Bikeshed's
linking text (with spaces) is ambiguous and would require a for=""
attribute.

What doesn't work:

  - Some cross-references as per above.
  - Some manual changes to HTML are still needed:
    https://github.com/whatwg/html/tree/bikeshed-experiment
  - Existing IDs aren't always preserved.
  - Simplification of HTML into Bikeshed shorthands like {{Node}}.
  - subdfn, which has no equivalent in Bikeshed.
  - Variants like the dev edition aren't properly supported.

freyjosh059-code commented on src/main.rs:
```diff
@@ -35,6 +43,12 @@ async fn run() -> io::Result<()> {
     // Find the paths we need.
     let cache_dir = path_from_env("HTML_CACHE", ".cache");
     let source_dir = path_from_env("HTML_SOURCE", "../html");
+    let use_bikeshed_str = env::var_os("USE_BIKESHED");
```
env::var_os("USE_BIKESHED")
